misc/testerina/modules/testerina-core/src/main/ballerina/execute.bal

```diff
@@ -374,6 +374,14 @@ function restructureTest(TestFunction testFunction, string[] descendants) return
 
     foreach function dependsOnFunction in testFunction.dependsOn {
         TestFunction dependsOnTestFunction = check testRegistry.getTestFunction(dependsOnFunction);
+
+        // if the dependsOnFunction is disabled, throw an error
+        if !dependsOnTestFunction.enabled {
```

Comment:
when does this fail?
Does this execute any tests before this point or fails at the beginning?

Reply:
It fails at the beginning before executing any tests